src/plonk/verifier.rs

```diff
@@ -261,12 +265,20 @@ impl<C: CurveAffine> Proof<C> {
         }
 
         // Verify the opening proof
-        self.opening.verify(
-            params,
-            &mut transcript,
-            x_6,
-            &f_commitment.to_affine(),
-            f_eval,
-        )
+        let guard = self
+            .opening
+            .verify(
+                params,
+                msm,
+                &mut transcript,
+                x_6,
+                &f_commitment.to_affine(),
+                f_eval,
+            )
+            .unwrap();
+
+        let msm_challenges = guard.use_challenges();
+
+        msm_challenges.is_zero()
```

Comment:
Instead, let's return the MSM so that the caller can use it for the next proof they verify.